moveit_planners/ompl/ompl_interface/src/detail/ompl_constraint.cpp

```diff
@@ -0,0 +1,74 @@
+#include "moveit/ompl_interface/detail/ompl_constraint.h"
+
+#include <moveit/robot_model/robot_model.h>
+#include <moveit/robot_state/robot_state.h>
+#include <moveit_msgs/Constraints.h>
+
+namespace ompl_interface
+{
+/****************************
+ * Base class for constraints
+ * **************************/
+
+BaseConstraint::BaseConstraint(robot_model::RobotModelConstPtr robot_model, const std::string& group,
+                               const unsigned int num_dofs, const unsigned int num_cons_)
+  : robot_model_(robot_model), ob::Constraint(num_dofs, num_cons_)
+{
+  // Setup Moveit's robot model for kinematic calculations
+  robot_state_.reset(new robot_state::RobotState(robot_model_));
+  robot_state_->setToDefaultValues();
+  joint_model_group_ = robot_state_->getJointModelGroup(group);
+
+  // use end-effector link by default
+  // \todo use the link specified in the constraint message
+  link_name_ = joint_model_group_->getLinkModelNames().back();
```

Comment:
By making the link name an optional constructor parameter you enable constraints on *any* link. There are variants of the getJacobian method that compute the Jacobian for a given link.

Reply:
Indeed, I still have to do this. I planned to do this by adding the following line to `parseConstraintMsg`
```
link_name_ = constraints.position_constraints.at(0).link_name;
```
so we go looking in the right place for the link name depending on what type of constraint is created.

This does not happen in the constructor because I had some issues with a dynamic binding that I could not get resolved without calling the `parseConstraintMsg` after the object has been constructed.

The latter happens in `void BaseConstraint::init(moveit_msgs::Constraints constraints)` which is another level of indirection that is probably uncessary.